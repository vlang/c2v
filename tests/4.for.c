#include <stdio.h>

void for_test() {
	for (int i = 0; i < 10; i++) {
		printf("i = %d\n", i);
	}
	for (int i = 0; i < 10; i++) printf("single line");
	//
	int x = 1;
	int sum = 0;
	while (x < 10) {
		x++;
		sum += x;
	}
	//
	while (1) {
		printf("inf loop");
		break;
	}
}
