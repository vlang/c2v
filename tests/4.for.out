[translated]
module main

fn for_test() {
	for i := 0; i < 10; i++ {
		C.printf(c'i = %d\n', i)
	}
	for i := 0; i < 10; i++ {
		C.printf(c'single line')
	}
	x := 1
	sum := 0
	for x < 10 {
		x++
		sum += x
	}
	for 1 {
		C.printf(c'inf loop')
		break
	}
}
